src/frontend/org/voltdb/utils/CatalogSizing.java

```
@@ -222,7 +222,10 @@ private static int getVariableColumnSize(int capacity, int dataSize, boolean for
             }
             bufferSize += increment;
         }
-        return bufferSize + 8 + 24;
+        // The rounded buffer size has an additional 4-byte allocation size and
+        // 8-byte back pointer overhead. There is also has an 8-byte pointer
```

A unit test could at least lock down a hard-coded value and then we would see a test failure if we ever introduced a bug into the code.  That's better than no test.

My tendency would be to avoid making any more product changes at this point so close to shipping 5.8.  It might make sense to make follow-up tickets for the refactoring you still want to do.